app/presenters/hyrax/admin_set_presenter.rb

```diff
@@ -3,5 +3,22 @@ class AdminSetPresenter < CollectionPresenter
     def total_items
       ActiveFedora::SolrService.count("{!field f=isPartOf_ssim}#{id}")
     end
+
+    # AdminSet cannot be deleted if default set or non-empty
+    def disable_delete?
+      total_items > 0 || default_set?
+    end
+
+    # Message to display if deletion is disabled
+    def disabled_message
+      return I18n.t('hyrax.admin.admin_sets.delete.error_default_set') if default_set?
+      return I18n.t('hyrax.admin.admin_sets.delete.error_not_empty') if total_items > 0
+    end
+
+    private
+
+      def default_set?
```

Comment:
Good question. I'm not sure I know how to access the model from this presenter?  Are there examples somewhere? Or do I have to reload the AdminSet via ID?